packages/workspace-utils/src/index.ts

```diff
@@ -65,7 +66,14 @@ export async function updateWorkspaceState(
   message = '',
 ): Promise<void> {
   // TODO: add locking
-  await queryAsync(sql.update_workspace_state, { workspace_id, state, message });
+  const delta = await queryRow(
+    sql.update_workspace_state,
+    { workspace_id, state, message },
+    IntervalSchema,
+  );
+  if (delta > 0) {
+    await updateCourseInstanceUsagesForWorkspace({ workspace_id, duration: delta });
```

Comment:
I'm reasonably confident that there's a units mismatch or something here. After launching a workspace on a fresh PL install, this recorded ~10 hours and 25 minutes of use. I'm guessing that `delta` is in milliseconds, but when inserting/adding that to an `interval` column, it's treated as seconds?

Reply:
Excellent catch, thank you! Fixed in ea2f606cee66392246ccc07474b059ebd762f98d